xbmc/utils/AMLUtils.cpp

```diff
@@ -216,20 +216,23 @@ bool aml_support_hevc_10bit()
   return (has_hevc_10bit == 1);
 }
 
-bool aml_support_h264_4k2k()
+int aml_support_h264_4k2k()
 {
   static int has_h264_4k2k = -1;
 
   if (has_h264_4k2k == -1)
   {
     std::string valstr;
     if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-    {
-      return false;
-    }
-    return (valstr.find("h264_4k2k:") != std::string::npos);
+      has_h264_4k2k = 0;
```

Comment:
Is it possible to introduce speaking constants here insted 0,1,2,3 ?
In general there will be always a point in the future where you are happy to not use such hardcoded numbers